### ISSUE TITLE
How is this project related to SAP Cloud SDK for JavaScript?

### ISSUE THREAD

**Issue by gregorwolf** [SAP/odata-library#31]
Hello odata-library maintainers,

I hope you are aware of the [SAP Cloud SDK for JavaScript](https://github.com/SAP/cloud-sdk-js) project. It does already provide funtionality for [OData](https://sap.github.io/cloud-sdk/docs/js/features/odata/overview) supporting V2 and V4. So my questions are:

- Is this project related or used in SAP Cloud SDK for JavaScript?
- If this project is not related why should I use it instead of the SAP Cloud SDK for JavaScript?

Best regards
Gregor

**Comment by mnezerka** [comment]
Hi Gregor. This project doesn't have any relation to mentioned SDK. I'll try to describe difference (as I understand it):
- SAP Cloud SDK is **generated** javascript that perfectly matches SAP Cloud OData APIs. As far as I know, generated library supports type checking and other features that were easily achievable by processing engine. Any change in SAP Cloud OData services (e.g. adding new API) requires generating new version of javascript SDK.
- OData Library approaches OData API in a different way. Everything happens in runtime. it fetches OData service description (metadata) and builds javascript bindings **dynamically**.

Nobody will force you to use OData Library or SAP Cloud SDK. It is solely up to you. The technical differences were explained in previous paragraph. Those are  points that could bring more light into use cases:
- SDK is **fixed** to SAP Cloud APIs, you cannot call any API in the world or even inside SAP. The initialisation could be easier since you don't need to know URL of the OData service (it is hardcoded in SDK js code). The SDK covers only published services. There are many cases where you need to call custom or internal OData API, which is not covered by SDK.
- OData Library is **generic** tool that should work with any OData API. So you can use it to call SAP Cloud OData APIs as well as any OData service in the world (e.g. public Northwind). You can also use this library for communication with many SAP services that are not covered by Cloud SDK. Many teams inside SAP use OData Library for testing purposes.

I hope it is a bit more clear now.

Michal

**Comment by gregorwolf** [comment]
Hi Michal,

thank you for the quick response. 

I think you also have the very common miss-understanding due to the name of the SAP Cloud SDK that it's only able to call SAP Cloud API. But that is wrong. I think the main reason why it's called SAP Cloud SDK is because of it's use in the SAP Business Technology Platform (FKA SAP Cloud Platform). 

Following the documentation [Generate a Client Using the Command Line Interface](https://sap.github.io/cloud-sdk/docs/js/features/openapi/generate-openapi-client/#generate-a-client-using-the-command-line-interface) a typed OData Client can be created for every OData V2 or V4 service.

Best  Regards
Gregor

**Comment by mnezerka** [comment]
Yes, you are right, it seems your can generate js wrappers for any OData service. It is long time since I checked javascript SAP Cloud SDK. I remember I tested very first versions of it and provided some feedback to the team. OData Library isn't new, it was developed for many years and become open source last year. And to be honest, the dynamic approach (without generating anything) is more preferred by our teams. We have also python sibling https://github.com/SAP/python-pyodata, which does the same as OData Library, but in python.
